repose-aggregator/components/filters/attribute-mapping-policy-validation-filter/src/main/scala/org/openrepose/filters/attributemapping/AttributeMappingPolicyValidationFilter.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.attributemapping
+
+import java.io.InputStream
+import javax.inject.Named
+import javax.servlet._
+import javax.servlet.http.HttpServletResponse.{SC_BAD_REQUEST, SC_UNSUPPORTED_MEDIA_TYPE}
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+import javax.xml.transform.stream.StreamSource
+import javax.xml.transform.{Source, TransformerException, TransformerFactory}
+
+import com.fasterxml.jackson.core.{JsonParseException, JsonProcessingException}
+import com.rackspace.identity.components.{AttributeMapper, XSDEngine}
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import net.sf.saxon.s9api.{SaxonApiException, XdmDestination}
+import org.openrepose.commons.utils.io.BufferedServletInputStream
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper
+
+import scala.util.{Failure, Success, Try}
+
+@Named
+class AttributeMappingPolicyValidationFilter extends Filter with LazyLogging {
+
+  import AttributeMappingPolicyValidationFilter._
+
+  // Create a transform factory using the default implementation and the classloader for this class
+  private val transformerFactory = TransformerFactory.newInstance(
+    "org.apache.xalan.xsltc.trax.TransformerFactoryImpl",
+    this.getClass.getClassLoader)
+
+  override def init(filterConfig: FilterConfig): Unit = {}
+
+  override def destroy(): Unit = {}
+
+  override def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
+    val httpServletRequest = request.asInstanceOf[HttpServletRequest]
+    val httpServletResponse = response.asInstanceOf[HttpServletResponse]
+    val requestContentType = httpServletRequest.getContentType
+
+    // Buffer the request payload so that we can validate the policy, but forward the original payload
+    val requestPayloadBuffer = new BufferedServletInputStream(httpServletRequest.getInputStream)
+    requestPayloadBuffer.mark(Integer.MAX_VALUE)
```

Comment:
This allows us to reset the stream after it is validated. That means that we aren't doing any normalization -- the reason being that it is non-trivial to accomplish for JSON requests, we already validate the policy on application (in the SAML filter), and that was not part of the acceptance criteria for the story. I admit that normalizing is probably the right way to go, but I figured we can iterate on what is here.